### ISSUE TITLE
[ci] Unit tests fail on Mac

### ISSUE THREAD

**Issue by jameslamb** [cpplint/cpplint#154]
Opening this issue from https://github.com/cpplint/cpplint/pull/149#issuecomment-646908591

While working on #149 , I found that the testing steps documented in [CONTRIBUTING.md](https://github.com/cpplint/cpplint/blob/develop/CONTRIBUTING.rst#development) seem to fail on Mac but pass on Linux. It's possible that the issue is about versions of some libraries and not operating system, but I have not been able to narrow that down.

## Reproducible Example

As of https://github.com/cpplint/cpplint/commit/010d807dfe4da782150a86817c60c73ecb7f1752, the most recent commit to `master` when I opened this issue, running this command:

```shell
pytest cpplint_unittest.py -k testInclude --no-cov
```

**Mac**

![image](https://user-images.githubusercontent.com/7608904/85971714-8ae7ce80-b993-11ea-9e09-f2517e145341.png)

> ============ 1 failed, 3 passed, 191 deselected, 1 warning in 0.79s =============

**Linux**

```shell
docker run \
    --volume $(pwd):/usr/local/src/cpplint \
    -it continuumio/miniconda3 \
    /bin/bash

cd /usr/local/src/cpplint
pip install .[dev]

pytest cpplint_unittest.py -k testInclude --no-cov
```

> ============================ 4 passed, 191 deselected in 0.37 seconds ============================

## Possible Issue

With some print debugging, I found that on my Mac laptop the first test inside `testIncludeItsHeader()` produces this output

> /private/var/folders/xq/wktq4zdx4jd3qdpk34d28m940000gn/T/tmp57wr40fh/test/foo.cc should include its header file /private/var/folders/xq/wktq4zdx4jd3qdpk34d28m940000gn/T/tmp57wr40fh/test/foo.h [build/include] [5]'

but that the string expected has this (identical except the leading /private

> /var/folders/xq/wktq4zdx4jd3qdpk34d28m940000gn/T/tmp57wr40fh/test/foo.cc should include its header file /var/folders/xq/wktq4zdx4jd3qdpk34d28m940000gn/T/tmp57wr40fh/test/foo.h [build/include] [5]

I'm unsure how that leading `/private` is getting into the path and why the tests wouldn't work on Mac.

## Note for contributors

It's worth noting these two facts:

*  `cpplint` itself works without issue on my Mac laptop, so I think the problem here may be limited to the tests
* `cpplint` is [currently only tested on Linux environments in its continuous integration](https://github.com/cpplint/cpplint/blob/010d807dfe4da782150a86817c60c73ecb7f1752/.travis.yml)

**Comment by tkruse** [comment]
So there is some symelinking going on: https://stackoverflow.com/questions/12482702/pythons-os-chdir-and-os-getcwd-mismatch-when-using-tempfile-mkdtemp-on-ma

Could you please try changing the line:
```
temp_directory = tempfile.mkdtemp()
```
to
```
temp_directory = os.path.realpath(tempfile.mkdtemp())
```
